src/gluonts/transform/feature.py

```diff
@@ -433,3 +433,198 @@ def map_transform(self, data: DataEntry, is_train: bool) -> DataEntry:
+
+    If `is_train=True` the lags have the same length as the `target` field.
+    If `is_train=False` the lags have length len(target) + pred_length
+
+    Parameters
+    ----------
+    target_field
+        Field with target values (array) of time series
+    output_field
+        Field name to use for the output.
+    pred_length
+        Prediction length.
+    base_freq
+        Base frequency, i.e., the frequency of the original time series.
+    agg_freq
+        Aggregate frequency, i.e., the frequency of the aggregate time series.
+    agg_lags
+        List of aggregate lags given in the aggregate frequncy. If some of them
+        are invalid (need some of the last `prediction_length` values to be computed)
+        they are ignored.
+    agg_fun
+        Aggregation function. Default is 'mean'.
+    rolling_agg:
+        Boolean indicating if the aggregation should be done in a centered rolling
+        window fashion (default) or by calendar dates.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        target_field: str,
+        output_field: str,
+        pred_length: int,
+        base_freq: str,
+        agg_freq: str,
+        agg_lags: List[int],
+        agg_fun: str = "mean",
+        rolling_agg: bool = True,
+        dtype: DType = np.float32,
+    ) -> None:
+        self.pred_length = pred_length
+        self.target_field = target_field
+        self.feature_name = output_field
+        self.base_freq = base_freq
+        self.agg_freq = agg_freq
+        self.agg_lags = agg_lags
+        self.agg_fun = agg_fun
+        self.rolling_agg = rolling_agg
+        self.dtype = dtype
+
+        self.ratio = pd.Timedelta(self.agg_freq) / pd.Timedelta(self.base_freq)
+        assert (
+            self.ratio.is_integer() and self.ratio >= 1
+        ), "The aggregate frequency should be a multiple of the base frequency."
+        self.ratio = int(self.ratio)
+
+        if rolling_agg:
+            self.half_window = (self.ratio - 1) // 2
+            self.valid_lags = [
+                x
+                for x in self.agg_lags
+                if x > (self.pred_length - 1 + self.half_window) / self.ratio
+            ]
+        else:
+            self.valid_lags = [
+                x
+                for x in self.agg_lags
+                if x > np.ceil((self.pred_length - 1) / self.ratio)
+            ]
+        if set(self.agg_lags) - set(self.valid_lags):
+            print(
+                f"The aggregate lags {set(self.agg_lags) - set(self.valid_lags)} "
+                f"of frequency {self.agg_freq} are ignored."
+            )
+
+    def map_transform(self, data: DataEntry, is_train: bool) -> DataEntry:
+        assert self.base_freq == data["start"].freq
+
+        # convert to pandas Series for easier indexing and aggregation
+        if is_train:
+            pd_ts = pd.Series(
+                data[self.target_field],
+                index=pd.date_range(
```

Comment:
If you find the transformation to be slow the pandas `date_range` may be a bottleneck. We could avoid this by computing a long date range once and then just indexing into this long array. 
Let me know if you have performance issues and want to look into this.

Reply:
I have not done large scale tests to observe this but I will keep it in mind if it is too slow and then we can discuss it.